application/controllers/survey_response.py

```diff
@@ -79,9 +82,14 @@ def _send_message_to_rabbitmq(encrypted_message, tx_id):
 
         log.info('Getting environmental details for rabbit')
         rabbitmq_amqp = current_app.config.get('RABBITMQ_AMQP')
-        rabbit_mq = RabbitMQSubmitter(rabbitmq_amqp)
+        publisher = QueuePublisher([rabbitmq_amqp], QUEUE_NAME)
+        try:
+            result = publisher.publish_message(encrypted_message, headers={"tx_id": tx_id},
+                                               immediate=False, mandatory=True)
+        except PublishMessageError:
+            return False
 
-        if rabbit_mq.send_message(encrypted_message, QUEUE_NAME, tx_id):
+        if result:
```

Comment:
Is there a need for this check? If the message does not get published it throws PublishMessageError and returns  false anyway.

Reply:
You're right. Just talked to James and when we have the `mandatory=True` flag on the publisher it throws an exception rather than returning False.